File: internal/api/v1/request/explorer.go
package request

// BulkUpdateTagsRequest represents a request to bulk update tags for multiple scenes
type BulkUpdateTagsRequest struct {
	SceneIDs []uint `json:"scene_ids" binding:"required,min=1"`
	TagIDs   []uint `json:"tag_ids"`
	Mode     string `json:"mode" binding:"required,oneof=add remove replace"`
}

// BulkUpdateActorsRequest represents a request to bulk update actors for multiple scenes
type BulkUpdateActorsRequest struct {
	SceneIDs []uint `json:"scene_ids" binding:"required,min=1"`
	ActorIDs []uint `json:"actor_ids"`
	Mode     string `json:"mode" binding:"required,oneof=add remove replace"`
}

// BulkUpdateStudioRequest represents a request to bulk update studio for multiple scenes
type BulkUpdateStudioRequest struct {
	SceneIDs []uint `json:"scene_ids" binding:"required,min=1"`
	Studio   string `json:"studio"`
}

// FolderSceneIDsRequest represents a request to get scene IDs in a folder
// Supports optional filters to get only IDs matching search criteria
type FolderSceneIDsRequest struct {
	StoragePathID uint     `json:"storage_path_id" binding:"required"`
	FolderPath    string   `json:"folder_path"`
	Recursive     bool     `json:"recursive"`
	Query         string   `json:"query"`
	TagIDs        []uint   `json:"tag_ids"`
	Actors        []string `json:"actors"`
	HasPornDBID   *bool    `json:"has_porndb_id"` // nil = no filter, true = has, false = missing
}

// BulkDeleteRequest represents a request to delete multiple videos
type BulkDeleteRequest struct {
	SceneIDs  []uint `json:"scene_ids" binding:"required,min=1"`
	Permanent bool   `json:"permanent"` // false = trash (default), true = permanent delete
}

// ScenesMatchInfoRequest represents a request to get minimal scene data for bulk matching
type ScenesMatchInfoRequest struct {
	SceneIDs []uint `json:"scene_ids" binding:"required,min=1"`
}

// FolderSearchRequest represents a request to search within a folder
type FolderSearchRequest struct {
	StoragePathID uint     `json:"storage_path_id" binding:"required"`
	FolderPath    string   `json:"folder_path"`
	Recursive     bool     `json:"recursive"`
	Query         string   `json:"query"`
	TagIDs        []uint   `json:"tag_ids"`
	Actors        []string `json:"actors"`
	Studio        string   `json:"studio"`
	MinDuration   int      `json:"min_duration"`
	MaxDuration   int      `json:"max_duration"`
	Sort          string   `json:"sort"`
	HasPornDBID   *bool    `json:"has_porndb_id"` // nil = no filter, true = has, false = missing
	Page          int      `json:"page"`
	Limit         int      `json:"limit"`
}

File: web/app/types/explorer.ts
import type { SceneListItem } from './scene';
import type { StoragePath } from './storage';

export interface FolderInfo {
    name: string;
    path: string;
    scene_count: number;
    total_duration: number;
    total_size: number;
}

export interface StoragePathWithCount extends StoragePath {
    scene_count: number;
}

export interface StoragePathsResponse {
    storage_paths: StoragePathWithCount[];
}

export interface FolderContentsResponse {
    storage_path: StoragePath;
    current_path: string;
    subfolders: FolderInfo[];
    scenes: SceneListItem[];
    total_scenes: number;
    page: number;
    limit: number;
}

export interface BulkUpdateTagsRequest {
    scene_ids: number[];
    tag_ids: number[];
    mode: 'add' | 'remove' | 'replace';
}

export interface BulkUpdateActorsRequest {
    scene_ids: number[];
    actor_ids: number[];
    mode: 'add' | 'remove' | 'replace';
}

export interface BulkUpdateStudioRequest {
    scene_ids: number[];
    studio: string;
}

export interface FolderSceneIDsRequest {
    storage_path_id: number;
    folder_path: string;
    recursive: boolean;
}

export interface BulkUpdateResponse {
    updated: number;
    requested: number;
}

export interface FolderSceneIDsResponse {
    scene_ids: number[];
    count: number;
}

export interface BulkDeleteRequest {
    scene_ids: number[];
    permanent?: boolean;
}

export interface BulkDeleteResponse {
    deleted: number;
    requested: number;
}

export interface FolderSearchRequest {
    storage_path_id: number;
    folder_path: string;
    recursive: boolean;
    query: string;
    tag_ids?: number[];
    actors?: string[];
    studio?: string;
    min_duration?: number;
    max_duration?: number;
    sort?: string;
    page: number;
    limit: number;
}

export interface FolderSearchResponse {
    scenes: SceneListItem[];
    total: number;
    page: number;
    limit: number;
}

export interface ScenesMatchInfoRequest {
    scene_ids: number[];
}

export interface SceneMatchInfo {
    id: number;
    title: string;
    original_filename: string;
    porndb_scene_id: string | null;
    actors: string[];
    studio: string | null;
    thumbnail_path: string;
    duration: number;
}

export interface ScenesMatchInfoResponse {
    scenes: SceneMatchInfo[];
}
